src/lib/formats/fdos_dsk.cpp

```diff
@@ -0,0 +1,185 @@
+// license:BSD-3-Clause copyright-holders:Michael R. Furman
+/*
+ * fdos_dsk.cpp  -  FDOS compatible disk images
+ *
+ * Created on: 24/08/2022
+ *
+ * This FDOS floppy disk image support leverages the wd177x_format support with
+ * verification of the disk formatting and the the ability to support format
+ * variations.
+ *
+ * The FDOS Disk format is as follows:
+ *
+ * Geometry:: 10 Sectors * 35 Tracks * 256 Bytes * 1 Side
+ * Sectors Numbered from 0-9
+ * Gap Bytes: Verified from FDOS source code
+ *
+ * Disk Format::
+ *
+ * Operating System:: Tracks 0-1
+ * Directory:: Track 2
+ * Data:: Tracks 3-34
+ *
+ *
+ * Note: Due to hardware differences between the original DC-1 (WD1771)
+ * controller for which FDOS was designed and DC-4 (WD1797) or DC-5 (WD2797)
+ * Controllers (DC-5 is Emulated in MAME), the machine can only be booted with
+ * a specific disk image from deramp.com (Mike Douglas) at the following
+ * location.  The other disks available in the same directory can then be read
+ * once FDOS is loaded. This disk image also works on real hardware with a DC-4
+ * or PT FD-2A controllers.
+ *
+ * https://deramp.com/downloads/swtpc/software/FDOS/Disk%20Images/FDOSMPS.DSK
+ *
+ * This disk contains the all of these required patches:
+ *
+ * https://deramp.com/downloads/swtpc/software/FDOS/Disk%20Images/Patches/
+ */
+
+#include "fdos_dsk.h"
+
+#include "imageutl.h"
+
+#include "ioprocs.h"
+
+namespace
+{
+	class fdos_formats : public wd177x_format
+	{
+	public:
+		struct dirent_entry_fdos
+		{
+			char filename[8]{};
+			char password[8]{};
+			uint8_t start_track = 0;
+			uint8_t start_sector = 0;
+			uint8_t num_sectors[2] = {};
+			uint8_t file_type = 0;
+			uint8_t start_addr[2] = {};
+			uint8_t end_addr[2] = {};
+			uint8_t exec_addr[2] = {};
+			uint8_t basic_high_line[2] = {};
+			uint8_t spares[3] = {};
```

Comment:
Why are you mixing `T foo[N]{};` and `T foo[N] = {};` style for array initialisation (with and without the equals sign) here?  Please pick a single style to use.

Reply:
Thanks for the catch here, your fix was what I actually intended.